Hast.Common/Models/VhdlHardwareDescription.cs

```diff
@@ -29,38 +30,31 @@ public class VhdlHardwareDescription : IHardwareDescription
         [JsonProperty]
         public IEnumerable<ITransformationWarning> Warnings { get; set; }
 
-
-        public async Task Serialize(Stream stream)
+        public async Task SerializeAsync(Stream stream)
         {
             if (string.IsNullOrEmpty(VhdlSource)) throw new InvalidOperationException("There is no VHDL source set.");
 
-            using (var writer = new StreamWriter(stream))
-            {
-                await writer.WriteAsync(JsonConvert.SerializeObject(
-                    this,
-                    Formatting.None,
-                    GetJsonSerializerSettings()));
-            }
+            using var writer = new StreamWriter(stream);
+            await writer.WriteAsync(JsonConvert.SerializeObject(
+                this,
+                Formatting.None,
+                GetJsonSerializerSettings()));
         }
 
-
-        public static async Task<VhdlHardwareDescription> Deserialize(Stream stream)
+        public static async Task<VhdlHardwareDescription> DeserializeAsync(Stream stream)
         {
-            using (var reader = new StreamReader(stream))
-            {
-                return JsonConvert.DeserializeObject<VhdlHardwareDescription>(
-                        await reader.ReadToEndAsync(),
-                        GetJsonSerializerSettings());
-            }
+            using var reader = new StreamReader(stream);
+            return JsonConvert.DeserializeObject<VhdlHardwareDescription>(
+                    await reader.ReadToEndAsync(),
+                    GetJsonSerializerSettings());
         }
 
-
         private static JsonSerializerSettings GetJsonSerializerSettings() =>
-            new JsonSerializerSettings
+            new()
             {
-                TypeNameHandling = TypeNameHandling.Auto,
+                TypeNameHandling = TypeNameHandling.None,

```

Comment:
Ez a sok warning miatt:
- VhdlHardwareDescription.cs(55, 36): [SCS0028] TypeNameHandling is set to the other value than 'None'. It may lead to deserialization vulnerability.
- VhdlHardwareDescription.cs(55, 36): [CA2326] Deserializing JSON when using a TypeNameHandling value other than None can be insecure.
- VhdlHardwareDescription.cs(53, 13): [CA2327] When deserializing untrusted input, allowing arbitrary types to be deserialized is insecure.  When using JsonSerializerSettings, use TypeNameHandling.None, or for values other than None, restrict deserialized types with a SerializationBinder.
- VhdlHardwareDescription.cs(47, 20): [CA2327] When deserializing untrusted input, allowing arbitrary types to be deserialized is insecure.  When using JsonSerializerSettings, use TypeNameHandling.None, or for values other than None, restrict deserialized types with a SerializationBinder.

Reply:
Hmm, érdekes. Viszont valszeg oka van, hogy ezt átállítottuk a defaultról. Kipróbáltad, jól működik a caching? (Ugye debugban ki van kapcsolva.)

Reply:
Ja okkal. Lepragmázom.